### PR TITLE
Util optimizations 

### PR DESCRIPTION
**Pull request by pizzi80**
Java 11+ optimizations

### CODE REVIEW

File: src/main/java/org/omnifaces/util/Utils.java [hunk 1]
```diff
@@ -270,9 +271,10 @@ public static boolean isBlank(String string) {
     public static boolean isNumber(String string) {
         try {
             // Performance tests taught that this approach is in general faster than regex or char-by-char checking.
-            return Long.valueOf(string) != null;
+            Long.parseLong(string);
+            return true;
         }
-        catch (Exception ignore) {
+        catch (NumberFormatException ignore) {
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java]: This doesn't cover NullPointerException.

File: src/main/java/org/omnifaces/util/Utils.java [hunk 2]
```diff
@@ -288,9 +290,10 @@ public static boolean isNumber(String string) {
     public static boolean isDecimal(String string) {
         try {
             // Performance tests taught that this approach is in general faster than regex or char-by-char checking.
-            return Double.valueOf(string) != null;
+            Double.parseDouble(string);
+            return true;
         }
-        catch (Exception ignore) {
+        catch (NumberFormatException ignore) {
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java]: This doesn't cover NullPointerException.

File: src/main/java/org/omnifaces/util/Utils.java [hunk 3]
```diff
@@ -288,9 +290,10 @@ public static boolean isNumber(String string) {
     public static boolean isDecimal(String string) {
         try {
             // Performance tests taught that this approach is in general faster than regex or char-by-char checking.
-            return Double.valueOf(string) != null;
+            Double.parseDouble(string);
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java:289]: Sonar will give a warning on this. Please restore `return ... != null`.

File: src/main/java/org/omnifaces/util/Utils.java [hunk 4]
```diff
@@ -776,8 +763,11 @@ public static <T> Stream<T> stream(Object object) {
         if (object == null) {
             return Stream.empty();
         }
+        if (object instanceof Collection) {
+            return ((Collection<T>)object).stream();
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java]: Try to be consistent with code format. Lack of blank space after cast operator deviates from existing code.

File: src/main/java/org/omnifaces/util/Utils.java [hunk 5]
```diff
@@ -645,9 +633,8 @@ else if (iterable instanceof Collection) {
         }
         else {
             List<E> list = new ArrayList<>();
-            Iterator<E> iterator = iterable.iterator();
-            while (iterator.hasNext()) {
-                list.add(iterator.next());
+            for (E e : iterable) {
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java]: Please write out variable names fully, `element` in this case.

File: src/main/java/org/omnifaces/util/Utils.java [hunk 6]
```diff
@@ -1239,12 +1224,7 @@ public static String decodeURL(String string) {
             return null;
         }
 
-        try {
-            return URLDecoder.decode(string, UTF_8.name());
-        }
-        catch (UnsupportedEncodingException e) {
-            throw new UnsupportedOperationException(ERROR_UNSUPPORTED_ENCODING, e);
```
Comment (BalusC) [inline src/main/java/org/omnifaces/util/Utils.java:1246]: Won't this leave behind an unused `ERROR_UNSUPPORTED_ENCODING` constant? If so then it should also be removed.

File: src/main/java/org/omnifaces/util/Utils.java (hunk 1 above)
Comment (pizzi80) [reply src/main/java/org/omnifaces/util/Utils.java]: The `Long.parseLong`, internally cover all the situations and throws a `NumberFormatException`, ... but for uniformity I'll bring back the generic `Exception'

File: src/main/java/org/omnifaces/util/Utils.java (hunk 3 above)
Comment (pizzi80) [reply src/main/java/org/omnifaces/util/Utils.java:289]: These new methods returns a primitive value, so we can't add a `... != null` check

**Pull request merged**